File: Arrays and String/189. Rotate Array/rotate_right.cpp
class Solution
{
public:
    void rotate(vector<int> &nums, int k)
    {
        int n = nums.size();
        int temp;
        k = k % n;

        for (int i = 0, j = n - k - 1; i < j; i++, j--)
        {
            temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        for (int i = n - k, j = n - 1; i < j; i++, j--)
        {
            temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }

        for (int i = 0, j = n - 1; i < j; i++, j--)
        {
            temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
    }
};

File: Arrays and String/189. Rotate Array/rotate_right.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k = k % n

        i = 0
        j = n - k - 1
        while i < j:
            temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
            i += 1
            j -= 1

        i = n - k
        j = n - 1
        while i < j:
            temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
            i += 1
            j -= 1

        i = 0
        j = n - 1
        while i < j:
            temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
            i += 1
            j -= 1
